elide-core/src/main/java/com/yahoo/elide/security/executors/ActivePermissionExecutor.java

```diff
@@ -111,7 +111,14 @@ public ActivePermissionExecutor(boolean verbose, final RequestScope requestScope
                                                                    ChangeSpec changeSpec) {
         Supplier<Expression> expressionSupplier = () -> {
             if (SharePermission.class == annotationClass) {
-                return expressionBuilder.buildSharePermissionExpressions(resource);
+                if (requestScope.getDictionary().isShareable(resource.getResourceClass())) {
+                    return expressionBuilder.buildAnyFieldExpressions(resource, ReadPermission.class, changeSpec);
+                } else {
+                    ForbiddenAccessException e = new ForbiddenAccessException("Missing SharePermission for "
```

Comment:
So the expressionSupplier just builds an expression - it doesn't actually execute it.  Inline with this code, this could build an expression that explicitly denies permission.

Later - executing that expression will throw the appropriate exception.